File: benchmarking/tasks/fib.c
#include <stdio.h>

int fib_rec(int n) {
    if (n <= 1) return n;
    return fib_rec(n - 1) + fib_rec(n - 2);
}

int fib_iter(int n) {
    if (n <= 1) return n;
    int a = 0, b = 1;
    for (int i = 2; i <= n; i++) {
        int temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

int main() {
    int n = 40;
    int result_rec = fib_rec(n);
    int result_iter = fib_iter(n);
    printf("fib_rec(%d): %d\n", n, result_rec);
    printf("fib_iter(%d): %d\n", n, result_iter);
    return 0;
}

File: benchmarking/tasks/fib.rs
fn fib_rec(n: u64) -> u64 {
    if n <= 1 {
        return n;
    }
    fib_rec(n - 1) + fib_rec(n - 2)
}

fn fib_iter(n: u64) -> u64 {
    if n <= 1 {
        return n;
    }
    let (mut a, mut b) = (0, 1);
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main() {
    let n: u64 = 40;
    let result_rec = fib_rec(n);
    let result_iter = fib_iter(n);
    println!("fib_rec({}): {}", n, result_rec);
    println!("fib_iter({}): {}", n, result_iter);
}
